gdx-jnigen/src/main/java/com/badlogic/gdx/jnigen/build/toolchains/GNUToolchain.java

```diff
@@ -0,0 +1,175 @@
+
+    private File cCompilerExecutable;
+    private File cppCompilerExecutable;
+    private File linkerExecutable;
+    private File stripperExecutable;
+
+    @Override
+    public void checkForTools () {
+        String cCompiler = target.compilerPrefix + target.cCompiler + target.compilerSuffix;
+        String cppCompiler = target.compilerPrefix + target.cppCompiler + target.compilerSuffix;
+        String linker = target.compilerPrefix + target.cppCompiler + target.compilerSuffix;
+        String stripper = target.compilerPrefix + "strip" + target.compilerSuffix;
+
+        //Check that these actually exist!
+        cCompilerExecutable = ToolFinder.getToolFile(cCompiler, ENV, ".exe", true);
+        cppCompilerExecutable = ToolFinder.getToolFile(cppCompiler, ENV, ".exe", true);
+        linkerExecutable = ToolFinder.getToolFile(linker, ENV, ".exe", true);
+
+        //Don't strip on Mac or in non release mode
+        stripperExecutable = ToolFinder.getToolFile(stripper, ENV, ".exe", target.release || target.os == Os.MacOsX);
+
+        logger.info("Toolchain is valid.\ncCompiler: {}\ncppCompiler: {}\nlinker: {}\nstripper: {}", cCompilerExecutable, cppCompilerExecutable, linkerExecutable, stripperExecutable);
+    }
+
+
+    @Override
+    public void compile () {
+        compileC();
+        compileCPP();
+
+        link();
+        strip();
+    }
+
+
+    private void compileC () {
+        if (sourceCollectedCFiles.isEmpty()) {
+            logger.info("No c files, skipping c compilation");
+            return;
+        }
+
+        List<String> args = new ArrayList<>();
+        args.addAll(stringFlagsToArgs(target.cFlags));
+        args.add("-Ijni-headers");
+        args.add("-Ijni-headers/" + target.os.getJniPlatform());
+        args.add("-I.");
+
+        for (String headerDir : target.headerDirs) {
+            args.add("-I" + convertPathToAbsolute(headerDir).getAbsolutePath());
+        }
+
+        ArrayList<String> perFileArgs = new ArrayList<String>();
+        perFileArgs.addAll(args);
+
+        for (File buildCFile : buildCFiles) {
+            perFileArgs.clear();
+            perFileArgs.addAll(args);
+
+            perFileArgs.add(buildCFile.getAbsolutePath());
+            perFileArgs.add("-o");
+            perFileArgs.add(new File(buildDirectory, toObjectFile(buildCFile.getName())).getAbsolutePath());
+
+            logger.info("Compiling C File {}", buildCFile.getAbsolutePath());
+            ToolchainExecutor.execute(cCompilerExecutable, config.jniDir.file(), perFileArgs, createToolChainCallback("C Compile"));
+        }
+    }
+
+    private void compileCPP () {
+        if (sourceCollectedCppFiles.isEmpty()) {
+            logger.info("No c files, skipping c compilation");
+            return;
+        }
+
+        List<String> args = new ArrayList<>();
+        args.addAll(stringFlagsToArgs(target.cppFlags));
+        args.add("-Ijni-headers/");
```

Comment:
We should make these absolute path too, since otherwise you can't re-run a build execution easily, if you are not in the "jni" dir.
Same goes for the other toolchains I think